### PR TITLE
clarify some comments in service discovery

### PR DESCRIPTION
**Pull request by ramaraochavali**
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/serviceregistry/serviceentry/servicediscovery.go [hunk 1]
```diff
@@ -348,22 +348,22 @@ func (s *ServiceEntryStore) serviceEntryHandler(_, curr config.Config, event mod
 		configsUpdated[makeConfigKey(svc)] = struct{}{}
 	}
 
-	// If service entry is deleted, cleanup endpoint shards for services.
```
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/serviceentry/servicediscovery.go:351]: why remove it

File: pilot/pkg/serviceregistry/serviceentry/servicediscovery.go [hunk 2]
```diff
@@ -348,22 +348,22 @@ func (s *ServiceEntryStore) serviceEntryHandler(_, curr config.Config, event mod
...
 	for _, svc := range deletedSvcs {
 		s.XdsUpdater.SvcUpdate(shard, string(svc.Hostname), svc.Attributes.Namespace, model.EventDelete)
 		configsUpdated[makeConfigKey(svc)] = struct{}{}
 	}
 
+	// If a service is updated and is not part of updatedSvcs, that means its endpoints might have changed.
+	// Trigger full push for DNS resolution ServiceEntry in such cases.
```
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/serviceentry/servicediscovery.go:357]: I would suggest comments from #23352, which is more clear

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/serviceentry/servicediscovery.go:357]: Not sure when that was changed

File: pilot/pkg/serviceregistry/serviceentry/servicediscovery.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/serviceregistry/serviceentry/servicediscovery.go:351]: updated to more comments

**Review by howardjohn — APPROVED** [review]

**Pull request merged**